### PR TITLE
(fleet) installer as a separate command

### PR DESCRIPTION
**Pull request by arbll**
This PR makes the daemon call the installer as a command instead of instantiating an installer internally.

### CODE REVIEW

File: cmd/installer/subcommands/installer/command.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+		Installer: i,
+		t:         newTelemetry(),
+		ctx:       ctx,
+		span:      span,
+	}
+}
+
+func newTelemetry() *telemetry.Telemetry {
+	apiKey := os.Getenv("DD_API_KEY")
+	site := os.Getenv("DD_SITE")
```
Comment (BaptisteFoy) [inline cmd/installer/subcommands/installer/command.go:66]: **❓ Question:** ‏Shouldn't we have a default?

File: cmd/installer/subcommands/installer/command.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	"github.com/DataDog/datadog-agent/cmd/installer/command"
+	"github.com/DataDog/datadog-agent/pkg/fleet/installer"
+	"github.com/DataDog/datadog-agent/pkg/fleet/telemetry"
+	"github.com/spf13/cobra"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Commands returns the installer subcommands.
+func Commands(_ *command.GlobalParams) []*cobra.Command {
```
[the remark this answers is not in the document]
Comment (BaptisteFoy) [reply cmd/installer/subcommands/installer/command.go:23]: file locking in the run path is probably enough

**Review by BaptisteFoy — APPROVED** [review]

File: cmd/installer/subcommands/installer/command.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (arbll) [reply cmd/installer/subcommands/installer/command.go:23]: need to add some form of mutex dpkg-style

**Review by raphaelgavache — APPROVED** [review]

File: pkg/fleet/installer/service/datadog_installer.go [hunk 3]
```diff
@@ -33,6 +37,58 @@ func SetupInstaller(ctx context.Context) (err error) {
...
+	}
+	err = os.MkdirAll("/opt/datadog-packages", 0755)
+	if err != nil {
+		return fmt.Errorf("error creating /opt/datadog-packages: %w", err)
+	}
+	err = os.MkdirAll("/var/log/datadog", 0755)
+	if err != nil {
+		return fmt.Errorf("error creating /var/log/datadog: %w", err)
+	}
+	err = os.MkdirAll("/var/run/datadog-packages", 0755)
```
Comment (raphaelgavache) [inline pkg/fleet/installer/service/datadog_installer.go:80]: locks dir needs to be 777

File: pkg/fleet/daemon/installer_exec.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+type installerCmd struct {
+	*exec.Cmd
+	span tracer.Span
+	ctx  context.Context
+}
+
+func (i *installerExec) newInstallerCmd(ctx context.Context, command string, args ...string) *installerCmd {
+	span, ctx := tracer.StartSpanFromContext(ctx, fmt.Sprintf("installer.%s", command))
```
[the remark this answers is not in the document]
Comment (arbll) [reply pkg/fleet/daemon/installer_exec.go:55]: noted as followup

**Pull request merged**